src/strings.bif

```diff
@@ -1366,9 +1366,14 @@ function ends_with%(str: string, sub: string%) : bool
 	if ( sub->Len() > str->Len() )
 		return zeek::val_mgr->Bool(false);
 
-	string s = str->ToStdString();
-	string sub_s = sub->ToStdString();
-	return zeek::val_mgr->Bool(s.rfind(sub_s) == (s.size() - sub_s.size()));
+	auto sub_s = sub->ToStdStringView();
+
+	// Create a string_view that only looks at the end of the string being searched
+	// with the same number of characters as the search string. This avoids possible
+	// pathological searches of big strings if the search string doesn't exist.
+	auto s = std::string_view{str->CheckString() + str->Len() - sub->Len(),
```

Comment:
Don't use `CheckString()` and instead  lso `ToStdStringView()`. `CheckString()` will 1) search the string for a `\0` from the beginning to the back for every invocation (most likely measurable impact for large strings) and 2) breaks `ends_with()` for Zeek strings with embedded '\0' characters due to replacing it with a placeholder string. It also tickles a reporter error message.

Suggest to add the following as a btest which works with Zeek 6.0 today:
```
# zeek -e 'print ends_with("a\x00\x00b", "\x00b")'
T
```

...

Hmm, `print "\x00" in "a\x00\x00b"` does tickle the reporter error and evaluates to `F`. From my view, there's nothing wrong with that and it should evaluate to `T`.

```
# zeek -e 'print "\x00" in "a\x00\x00b";'
error in <command line>, line 3: string with embedded NUL: "\x00"
F
```

Oh, and this is `T`, obviously :-)
```
# ZEEK_ALLOW_INIT_ERRORS=1 zeek -e 'print "aaa\x00bbb" in "<string-with-NUL>"'
error in <command line>, line 3: string with embedded NUL: "aaa\x00bbb"
T
```